object-filter/src/main/java/org/infinispan/objectfilter/impl/BaseMatcher.java

```diff
@@ -22,7 +20,8 @@
  * @author anistor@redhat.com
  * @since 7.0
  */
-abstract class BaseMatcher<TypeMetadata, AttributeMetadata, AttributeId extends Comparable<AttributeId>> implements Matcher {
+//todo [anistor] make package local
```

Comment:
@danberindei Looks like @anistor plan to hide this in the future, since atm it is used by the query module, so the comment should stay